volatility3/framework/symbols/windows/extensions/consoles.py

```diff
@@ -0,0 +1,439 @@
+            return exe_name.get_string()
+
+        if self.ExeName.Length < 8:
+            return self.ExeName.Chars.cast(
+                "string",
+                encoding="utf-16",
+                errors="replace",
+                max_length=self.ExeName.Length * 2,
+            )
+        elif self.ExeName.Length < 1024:
+            return self.ExeName.Pointer.dereference().cast(
+                "string", encoding="utf-16", errors="replace", max_length=512
+            )
+
+    def get_aliases(self):
+        """Generator for the individual aliases for a
+        particular executable."""
+        for alias in self.AliasList.to_list(
+            f"{self.get_symbol_table_name()}{constants.BANG}_ALIAS",
+            "ListEntry",
+        ):
+            yield alias
+
+
+class SCREEN_INFORMATION(objects.StructType):
+    """A Screen Information Structure."""
+
+    @property
+    def ScreenX(self):
+        # 22000 change from an array of pointers to _ROW to an array of _ROW
+        row = self.TextBufferInfo.BufferRows.Rows[0]
+        if hasattr(row, "Row"):
+            return row.Row.RowLength2
+        else:
+            return row.RowLength2
+
+    @property
+    def ScreenY(self):
+        return self.TextBufferInfo.BufferCapacity
+
+    def _truncate_rows(self, rows):
+        """To truncate empty rows at the end, walk the list
+        backwards and get the last non-empty row. Use that
+        row index to splice. Rows are created based on the
+        length given in the ROW structure, so empty rows will
+        be ''."""
+
+        non_empty_index = 0
+        rows_traversed = False
+
+        for index, row in enumerate(reversed(rows)):
+            # the string was created based on the length in the ROW structure so it shouldn't have any bad data
+            if len(row.rstrip()) > 0:
+                non_empty_index = index
+                break
+            rows_traversed = True
+
+        if non_empty_index == 0 and rows_traversed:
+            rows = []
+        else:
+            rows = rows[0 : len(rows) - non_empty_index]
+
+        return rows
+
+    def get_buffer(self, truncate_rows=True, truncate_lines=True):
+        """Get the screen buffer.
+
+        The screen buffer is comprised of the screen's Y
+        coordinate which tells us the number of rows and
+        the X coordinate which tells us the width of each
+        row in characters. Windows 10 17763 changed from
+        a large text buffer to a grid of cells, with each
+        cell containing a single wide character in that
+        cell, stored in a CharRowCell object.
+
+        @param truncate: True if the empty rows at the
+        end (i.e. bottom) of the screen buffer should be
+        supressed.
+        """
+        rows = []
+
+        capacity = self.TextBufferInfo.BufferCapacity
+        start = self.TextBufferInfo.BufferStart
+        buffer_rows = self.TextBufferInfo.BufferRows.dereference()
```

Comment:
It's not clear why so many pointers are being explicitly dereferenced?  It should be that Self.TextBufferInfo.BufferRows.Rows will do the dereferencing for you.  I'd rather not litter the code with unnecessary `dereference`s in case people come to use it as a teaching guide...

Reply:
worked without it, so removed